numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -26,6 +26,7 @@ static PyObject *
 get_array_function(PyObject *obj)
 {
     static PyObject *ndarray_array_function = NULL;
+    static PyObject *array_function;
```

Comment:
This should not be static
```suggestion
    PyObject *array_function;
```

Reply:
sorry, I don't know too much about C, so I just copied the declaration of `ndarray_array_function`

Reply:
`static` means the variable is at a single location in memory, meaning that if multiple threads call this function at the same time, they can clobber each other. Python's Global Interpreter lock might mean this can't actually happen, but better safe than sorry. Thanks for the fix!